### PR TITLE
add support for --log-fmt=nprint

### PR DESCRIPTION
**Pull request by stonea**
Often times I want to use `--log` to get pretty prints of the AST after each pass but I want it in the same format used by `nprint_view` which I use when debugging.

This PR adds a new developer-oriented flag `--log-fmt <format>` where (for now) format can either be `default` or `nprint`. If left off, as one would expect, it defaults to `default` and we print logs the same as we always have.

In terms of implementation, to get this to work I ended up redirecting `stdout` to the log file we're generating.  This comment I added kind of explains it:

```C++
  // This is an absolute hack. nprint_view prints directly to stdout, we should
  // update it to take a file pointer use it and pass it along as needed. But rather
  // than update all the code I instead redirect stdout to the file (and then restore it
  // after the dump is done).  This makes debugging a pain and folks can really get caught
  // off-gaurd when stdout is no longer really stdout.
```

But I'm going to justify this as:
* This functionality is useful, and
* enabling it even in this hackish way is moving us in the right direction.

### CODE REVIEW

File: compiler/AST/AstDump.cpp [hunk 1]
```diff
@@ -81,8 +84,25 @@ void AstDump::view(const char* passName, int passNum) {
       AstDump logger;
 
       if (logger.open(module, passName, passNum) == true) {
-        module->accept(&logger);
-        logger.close();
+        if(fLogFormat == LogFormat::NPRINT) {
+          // This is an absolute hack. nprint_view prints directly to stdout, we should
+          // update it to take a file pointer use it and pass it along as needed. But rather
+          // than update all the code I instead redirect stdout to the file (and then restore it
```
Comment (e-kayrakli) [inline compiler/AST/AstDump.cpp:90]: Can we not make this happen by adding an optional argument to `nprint_view`?

Comment (e-kayrakli) [reply compiler/AST/AstDump.cpp:90]: If a quick try doesn't deliver results, you can merge the PR as-is. Ideally, a private issue could be nice as a rainy-day task for the future.

**Review by e-kayrakli — APPROVED** [review]

File: compiler/AST/AstDump.cpp (hunk 1 above)
Comment (stonea) [reply compiler/AST/AstDump.cpp:90]: > Can we not make this happen by adding an optional argument to nprint_view?

You mean passing in a FILE* or id as an optional arg (and have it default to stdout)? Yes I agree, that's what we would want to do. We'd then have to update all calls to `printf` to `fprintf` or whatever.

I actually don't think it'd be that bad, could be done in less-than a day, but it's more than a few minutes of work. I agree would be a nice rainy-day task. I'll create an issue to follow up here: https://github.com/Cray/chapel-private/issues/5608

Comment (e-kayrakli) [reply compiler/AST/AstDump.cpp:90]: Oh, I thought you were shying away from updating all the calls to `nprint_view` itself, and that got me thinking "why not an optional argument?". Ok that makes sense.

**Pull request merged**